src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexNode.cs

```diff
@@ -2001,7 +1997,7 @@ internal SymbolicRegexNode<TSet> PruneAnchors(SymbolicRegexBuilder<TSet> builder
         {
             //first prune the anchors in the node
             TSet wlbSet = builder._wordLetterForBoundariesSet;
-            TSet startSet = GetStartSet(builder);
+            TSet startSet = _set;
```

Comment:
Do we need the local anymore, or can the two uses of startSet below just be _set?